samples/unit-tests/series/datagrouping/demo.js

```diff
@@ -748,3 +748,44 @@ QUnit.test('DataGrouping and update', function (assert) {
         (#11471)`
     );
 });
+
+QUnit.test('Panning with dataGrouping and ordinal axis, #3825.', function (assert) {
+    const chart = Highcharts.stockChart('container', {
+        xAxis: {
+            ordinal: true
+        },
+        rangeSelector: {
+            selected: 0
+        },
+        series: [{
+            data: usdeur,
+            dataGrouping: {
+                forced: true,
+                groupAll: true,
+                units: [['day', [1]]]
+            }
+        }]
+    });
+    // Call function responsible for calculating positions of invisible points while panning.
+    chart.xAxis[0].ordinal.getExtendedPositions();
+
+    const positions = chart.xAxis[0].ordinal.positions,
+        positionsLength = positions.length,
+        index = chart.xAxis[0].ordinal.index,
+        indexArray = index[Object.keys(index)[0]],
+        indexLength = indexArray.length,
+        splicedIndex = // get data for current extremes
+            indexArray.splice(indexLength - positionsLength, positionsLength),
+        areEqual = positions.every(function (element, index) {
+            return element === splicedIndex[index];
+        });
+
+    assert.ok(
```

Comment:
Instead of `assert.ok(...)` you can use `assert.deepEqual(positions, splicedIndex)` (in case of error, it will show more information, for example which values don't match. `ok()` will just show true/false)